src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -144,6 +144,18 @@ def _get_extension_instance_name(instance_view, publisher, extension_type_name,
     return extension_instance_name
 
 
+# separated for aaz based implementation
+def _get_extension_instance_name1(instance_view, publisher, extension_type_name,
+                                  suggested_name=None):
```

Comment:
It is better to have a more eady-to-understand name.
```suggestion
def _get_extension_instance_name_from_aaz_instance(instance_view, publisher, extension_type_name,
                                  suggested_name=None):
```